### PR TITLE
Origin/secplugs/dev

### PR DESCRIPTION
**Pull request by Ajaatsecplugs**
---
Please check the following conditions for your PR.

- [x] `apiDefinition.swagger.json` is validated using `paconn validate` command.
- [x] `apiProperties.json` has a valid brand color. Invalid brand colors are `#007ee5` and `#ffffff`.

### CODE REVIEW

**Comment by ptalrejagithub** [comment]
@Ajaatsecplugs please provide readme.md.

**Comment by sriyen-msft** [comment]
@Ajaatsecplugs Please sign the CLA to get the PR reviewed

**Comment by Ajaatsecplugs** [comment]
Thank you.. I have signed the CLA.. Will submit readme shortly..

thanks
Shankar

**Comment by Ajaatsecplugs** [comment]
@sriyen-msft @ptalrejagithub please have a look in PR, we have added readme.md.

File: custom-connectors/secplugs/apiDefinition.swagger.json [hunk 1]
```diff
@@ -0,0 +1,445 @@
...
+              "properties": {
+                "score": {
+                  "type": "integer",
+                  "format": "int32",
+                  "description": "score"
+                },
+                "json_report": {
+                  "type": "array",
+                  "items": {},
+                  "description": "json_report"
```
Comment (sriyen-msft) [inline custom-connectors/secplugs/apiDefinition.swagger.json:37]: @Ajaatsecplugs Please provide valid description value as it seems like an operation ID.

Documentation - [Summary and Description Values](https://docs.microsoft.com/connectors/custom-connectors/certification-submission#operation-and-parameter-summaries-and-descriptions)

Applies to all the description values

File: custom-connectors/secplugs/apiDefinition.swagger.json [hunk 2]
```diff
@@ -0,0 +1,445 @@
...
+          }
+        },
+        "summary": "Scan files using SecPlug Scan",
+        "operationId": "filescan",
+        "parameters": [
+          {
+            "in": "header",
+            "name": "x-api-key",
+            "type": "string",
+            "default": "GW5sb8sj8D9CtvVrjsTC22FNljxhoVuL1UoM6fFL",
```
Comment (sriyen-msft) [inline custom-connectors/secplugs/apiDefinition.swagger.json:160]: @Ajaatsecplugs It looks like a key or secret which are not allowed to mention in any connector artifacts due to security concerns.

File: custom-connectors/secplugs/apiDefinition.swagger.json [hunk 3]
```diff
@@ -0,0 +1,445 @@
...
+                "data",
+                "filename",
+                "cte",
+                "filetype"
+              ]
+            }
+          }
+        ],
+        "description": "Scan files using SecPlug Scan",
+		"consumes": ["application/json"]
```
Comment (sriyen-msft) [inline custom-connectors/secplugs/apiDefinition.swagger.json:216]: @Ajaatsecplugs Please fix indentation issues in all the artifacts

File: custom-connectors/secplugs/apiProperties.json [hunk 4]
```diff
@@ -0,0 +1,8 @@
+{
+  "properties": {
+    "connectionParameters": {},
```
Comment (sriyen-msft) [inline custom-connectors/secplugs/apiProperties.json:3]: @Ajaatsecplugs The connectionParameters property has mentioned with empty body in apiProiperties.json file. Empty connection object will be created automatically without any input from an user. It is less secure to use empty connection parameters. Please update with the right definition or let us know the reason if you have chosen for "no authentication".

File: custom-connectors/secplugs/apiProperties.json [hunk 5]
```diff
@@ -0,0 +1,8 @@
...
+    "iconBrandColor": "#808080",
+    "capabilities": [],
+    "publisher": "secplugs"
```
Comment (sriyen-msft) [inline custom-connectors/secplugs/apiProperties.json:6]: @Ajaatsecplugs Please add `stackOwner` property as well..

Documentation - [Adding metadata](https://docs.microsoft.com/connectors/custom-connectors/certification-submission#adding-metadata)

File: custom-connectors/secplugs/settings.json [hunk 6]
```diff
@@ -0,0 +1,9 @@
+{
```
Comment (sriyen-msft) [inline custom-connectors/secplugs/settings.json:1]: @Ajaatsecplugs Please remove this file, as it is not required for open sourcing or certifying your connector

**Review by sriyen-msft — CHANGES_REQUESTED** [review]

File: custom-connectors/secplugs/readme.md [hunk 7]
```diff
@@ -0,0 +1,30 @@
+
+### NOTE
+> This is a Secplug scan connector.  The connector is provided here with the intent to illustrate certain features and functionality around connectors.
+
+## Secplug Scan Connector
+Secplug provides a powerful pltform for pluggable security using secplug plugins.
```
Comment (ptalrejagithub) [inline custom-connectors/secplugs/readme.md:6]: should be platform, spelling issue.

File: custom-connectors/secplugs/apiDefinition.swagger.json [hunk 8]
```diff
@@ -0,0 +1,445 @@
...
+    "title": "Secplug Scan",
+    "description": "Secplug Scan for scanning Emails, files and Urls",
+    "version": "1.0",
+	"contact": {
+      "name": "Secplugs",
+      "url": "https://secplugs.com/",
+      "email": "secplugs@secplugs.com"
+    }	
+  },
+  "host": "proxy.secplugs.com",
```
[the remark this answers is not in the document]
Comment (Ajaatsecplugs) [reply custom-connectors/secplugs/apiDefinition.swagger.json:13]: Yes its valid host url.

File: custom-connectors/secplugs/apiDefinition.swagger.json (hunk 2 above)
Comment (Ajaatsecplugs) [reply custom-connectors/secplugs/apiDefinition.swagger.json:160]: this is an open default key, available for public.

File: custom-connectors/secplugs/apiProperties.json (hunk 4 above)
Comment (Ajaatsecplugs) [reply custom-connectors/secplugs/apiProperties.json:3]: we have chosen no for authentication.

File: custom-connectors/secplugs/readme.md [hunk 9]
```diff
@@ -0,0 +1,30 @@
...
+
+
+## Prerequisites
+You will need the following to proceed:
+* A Microsoft Power Apps or Power Automate plan with custom connector feature
+* An account in secplugs
+* The Power platform CLI tools
+
+## Building the connector 
+you would need to ensure that you have an account in secplug and using https://secplugs.com/index.php/providers an power automate profile is created.
```
Comment (ptalrejagithub) [inline custom-connectors/secplugs/readme.md:17]: edit the "you" to You.

**Comment by Ajaatsecplugs** [comment]
@ptalrejagithub @sriyen-msft we have taken care of all review comments, please have a look.

File: custom-connectors/secplugs/apiDefinition.swagger.json [hunk 10]
```diff
@@ -0,0 +1,445 @@
...
+                },
+                "threat_object": {
+                  "type": "object",
+                  "properties": {
+                    "sha256": {
+                      "type": "string",
+                      "description": "sha256 of file content"
+                    }
+                  },
+                  "description": "threat_object"
```
Comment (sriyen-msft) [inline custom-connectors/secplugs/apiDefinition.swagger.json:47]: @Ajaatsecplugs Can you please provide valid description for `threat_object`

File: custom-connectors/secplugs/apiDefinition.swagger.json (hunk 2 above)
Comment (sriyen-msft) [reply custom-connectors/secplugs/apiDefinition.swagger.json:160]: @Ajaatsecplugs Is it possible to remove it form the code

File: custom-connectors/secplugs/apiDefinition.swagger.json [hunk 11]
```diff
@@ -0,0 +1,445 @@
...
+                  "type": "integer",
+                  "format": "int32",
+                  "description": "score of email scanned"
+                },
+                "json_report": {
+                  "type": "object",
+                  "properties": {
+                    "malwarereport": {
+                      "type": "string",
+                      "description": "malwarereport"
```
Comment (sriyen-msft) [inline custom-connectors/secplugs/apiDefinition.swagger.json:237]: @Ajaatsecplugs Can it be `malware report` or `Malware Report` ?

File: custom-connectors/secplugs/apiDefinition.swagger.json [hunk 12]
```diff
@@ -0,0 +1,445 @@
...
+          }
+        },
+        "summary": "Scan Emails using SecPlug Scan",
+        "operationId": "emailscan",
+        "parameters": [
+          {
+            "in": "header",
+            "name": "x-api-key",
+            "type": "string",
+            "default": "GW5sb8sj8D9CtvVrjsTC22FNljxhoVuL1UoM6fFL",
```
Comment (sriyen-msft) [inline custom-connectors/secplugs/apiDefinition.swagger.json:373]: @Ajaatsecplugs Can it be removed from the code ?

File: custom-connectors/secplugs/apiProperties.json [hunk 13]
```diff
@@ -0,0 +1,9 @@
+{
+  "properties": {
+    "connectionParameters": {},
+    "iconBrandColor": "#808080",
```
Comment (sriyen-msft) [inline custom-connectors/secplugs/apiProperties.json:4]: @Ajaatsecplugs Please make sure that you are following instructions for [icon](https://docs.microsoft.com/en-us/connectors/custom-connectors/certification-submission#icon) and `iconBrandColor` is actually the background color of your connector's icon.

**Review by sriyen-msft — CHANGES_REQUESTED** [review]

File: custom-connectors/secplugs/apiProperties.json (hunk 13 above)
Comment (Ajaatsecplugs) [reply custom-connectors/secplugs/apiProperties.json]: fixed

**Review by sriyen-msft — APPROVED** [review]

**Pull request merged**